iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/service/thrift/ConfigNodeRPCServiceProcessor.java

```diff
@@ -244,6 +246,24 @@ public TGetClusterIdResp getClusterId() {
 
   @Override
   public TDataNodeRegisterResp registerDataNode(TDataNodeRegisterReq req) {
+    // If the remote server is configured to bind to all ip addresses (0.0.0.0),
+    // update the request to use the remote address we got this request from.
+    if ((req.dataNodeConfiguration.getLocation() != null)
+        && req.dataNodeConfiguration.getLocation().isSetClientRpcEndPoint()
+        && req.dataNodeConfiguration.getLocation().getClientRpcEndPoint().isSetIp()
+        && "0.0.0.0"
```

Comment:
In the current version, we allow an RPCAddress of 0.0.0.0 to join the cluster; for example, the default RPCAddress for 1c1d is 0.0.0.0.
Here we don't have to do this replacement, we can support this node to join, because we can use internalAddress to remove the node without any problems

Reply:
Agree

Reply:
We are not preventing the node from joining at all ... we are just ensuring we are using a real IP address to list the node ... the only difference is, that in this case the data-node will probably use 127.0.0.1 as address.